modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        String name = user.getUserName();
+
+        return new Object[][]{
+                {id, name, null, null, ResponseCodeConstants.CODE_OK, user},
+                {id, name, null, "emails", ResponseCodeConstants.CODE_OK, user},
+                {id, name, "username,meta", null, ResponseCodeConstants.CODE_OK, user},
+                {id, name, "username", "emails", ResponseCodeConstants.CODE_OK, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetSuccess")
+    public void testGetSuccess(String id, String name, String attributes,
+                               String excludeAttributes, int expectedScimResponseStatus, Object objectUser)
+            throws BadRequestException, NotFoundException, CharonException {
+
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getMe(name, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        //Assertions
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        Assert.assertEquals(obj.getString("userName"), name);
+        Assert.assertEquals(obj.getString("id"), id);
+
+        String returnedURI = outputScimResponse.getHeaderParamMap().get(SCIMConstants.LOCATION_HEADER);
+        String expectedURI = ENDPOINT_ME + "/" + obj.getString("id");
+        Assert.assertEquals(returnedURI, expectedURI);
+
+        if (attributes == null & excludeAttributes == null) {
+
+            Assert.assertTrue(obj.has("emails"));
+
+        } else if (attributes == null & excludeAttributes != null) {
+
+            Assert.assertFalse(obj.has("emails"));
+
+        }
+
+    }
+
+    @DataProvider(name = "dataForGetNotFoundException")
+    public Object[][] dataToGetNotFoundException() {
+
+        return new Object[][]{
+                {"David", "userName", null, ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND}
```

Comment:
no need to have `ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND` in the data provider.

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d